### ISSUE TITLE
Scoring Reftests

### ISSUE THREAD

**Issue by fantasai** [web-platform-tests/wpt.fyi#3435]
I was drafting up advice for when to split out reftests into multiple files or gather them into a single one, and one concern that came in was “combining reftests reduces scores in WPT, which because devrel uses WPT scores might not be ideal”, which... is not great. :/ Can we make this concern go away somehow?

I would like to be able to give advice based on technical arguments like “significantly improves understandability of the tests” and not ask people to worry about WPT test scores.

**Comment by fantasai** [comment]
Actually splitting out the scoring for reftests seems hard, but, maybe we can weight them differently? "This test is really 12 tests." Or list them out with descriptions, similar to testharness.js descriptions, but without the ability to report individual scores? That would maybe also encourage people to comment on what exactly they're testing...

**Comment by KyleJu** [comment]
Let me bring it up in our WPT sync meetings and see if there best practices for this!

**Comment by KyleJu** [comment]
@fantasai 
> combining reftests affects scores in WPT

The short answer to this question is yes. It will affect WPT scores, and also depends on how many reftests and which reftests are changed. The exact detail is worth being discussed at the wpt infra meetings. Would you mind filing a similar bug in https://github.com/web-platform-tests/interop/issues so it can be brought up in future meetings?

**Comment by fantasai** [comment]
>> combining reftests affects scores in WPT
> The short answer to this question is yes.

That wasn't a question. 

> It will affect WPT scores

That is the premise under which this issue was filed, yes.

> Would you mind filing a similar bug in https://github.com/web-platform-tests/interop/issues so it can be brought up in future meetings?

Would you mind explaining what you think this issue is about or what kind of action by Interop you think would fix it? Because I'm not sure I understand why you want to transfer it to Interop.

**Comment by KyleJu** [comment]
> > > combining reftests affects scores in WPT
> > 
> > 
> > The short answer to this question is yes.
> 
> That wasn't a question.
> 
> > It will affect WPT scores
> 
> That is the premise under which this issue was filed, yes.
> 
> > Would you mind filing a similar bug in https://github.com/web-platform-tests/interop/issues so it can be brought up in future meetings?
> 
> Would you mind explaining what you think this issue is about or what kind of action by Interop you think would fix it? Because I'm not sure I understand why you want to transfer it to Interop.

My bad, I should have explained a bit more. Your question is around understanding the scoring of reftests after proposed actions. In that case, the interop group will have better expertise for it. They have a weekly meeting about WPT infra (not just strictly interop) that discusses a range of topics. @jcscottiii Just to confirm with you, do you think the interop repo is a good place to transfer this issue?

**Comment by jcscottiii** [comment]
Yes, [interop](https://github.com/web-platform-tests/interop ) is a better location for this

**Comment by fantasai** [comment]
> Your question is around understanding the scoring of reftests after proposed actions.

Sorry, I must have been entirely incomprehensible in my original posting. I'm not asking a question in order to clarify my understanding of reftest scoring. I understand how they are scored. I have a **problem** with how they are scored, and I am asking for that problem to be fixed.

Specifically, I'm asking wpt.fyi to be able to score reftest files that contain multiple tests as having multiple tests. Doing this will require such reftests to include some additional information detailing what or at least how many subtests they contain, and it will require wpt.fyi to be able to process that information.

The reason I'm asking for this is that people writing tests are suggesting that wpt.fyi scoring should be a consideration in how the tests are structured, and _I don't want this to be a consideration_. The choice of whether to use reftests or testharness.js test, or whether to structure reftests into one file or multiple files, should be based on the understandability of the test and its code, and on whether the feature is more comprehensively tested via reftest or testharness methods, and not on whether it will improve or detract from any individual browser's wpt.fyi score.

Is what I'm asking for making sense now?
